app/services/publisher_wallet_getter.rb

```diff
@@ -140,6 +147,36 @@ def total_balance_bat(accounts)
     accounts.map {|account| account["balance"].to_d }.reduce(0, :+)
   end
 
+  def last_settlement_balance(transactions)
+    return {} if transactions == []
+    last_settlement_date = transactions.reduce(Date.new(0)) { |last_settlement_date, transaction|
```

Comment:
This is convoluted from a readability standpoint. Would you simplify it?

Reply:
I could do this this imperatively with a loop
```ruby
last_settlement_date = Date.new(0)
transactions.each do |transaction|
  if transaction["created_at"].to_date > last_settlement_date && transaction["settlement_amount"].present?
    last_settlement_date = transaction["created_at"].to_date
  end
end
```

or I could separate the logic that selects only settlement transactions

```ruby
last_settlement_date = transactions.select { |transactions|
  transaction["settlement_amount"].present?
}.reduce(Date.new(0)) { |last_settlement_date, transaction|
  if transaction["created_at"].to_date > last_settlement_date
    last_settlement_date = transaction["created_at"].to_date
  end
  last_settlement_date
}
```

or I could do a little bit of both

```ruby
transactions = transactions.select { |transactions|
  transaction["settlement_amount"].present?
}

last_settlement_date = Date.new(0)
transactions.each do |transaction|
  if transaction["created_at"].to_date > last_settlement_date
    last_settlement_date = transaction["created_at"].to_date
  end
end

```

Let me know what you'd prefer @yachtcaptain23

Reply:
you might be able to cheat a bit since transactions is ordered by `created_at` when returned from eyeshade

maybe something like this?
```
last_settlement = transactions.lazy.reverse_each().select{|t| 
  t["settlement_amount"].present?
}.map{|t| t["created_at"]}.first() || Date.new(0)
```